Initial troubleshooting guide (#3640)
EnMasseProject/enmasse: documentation/modules/proc-troubleshooting.adoc

[id='troubleshooting-{context}']

> = Find which pods belong to an address space

Reviewer: ```suggestion
= Determine which pods belong to an address space
```

.Procedure

> . Retrieve the uuid for the address space

Reviewer: ```suggestion
. Retrieve the uuid for the address space:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} get addressspace _myspace_ -o jsonpath='{.metadata.annotations.enmasse\.io/infra-uuid}'
----

> . Find all pods with matching `infraUuid` label:

Reviewer: ```suggestion
. Find all pods with a matching `infraUuid` label:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} get pods -l infraUuid=_uuid_
----

> = Find currently running version of {ProductName}

Reviewer: ```suggestion
= Determine the currently running version of {ProductName}
```

.Procedure

> . Find enmasse-operator pod name:

Reviewer: ```suggestion
. Determine the `enmasse-operator` pod name:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} get pods | grep enmasse-operator
----

> . Print the VERSION environment of the enmasse-operator pod:

Reviewer: ```suggestion
. Print the `VERSION` environment of the `enmasse-operator` pod:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} rsh _enmasse-operator-...._ env | grep VERSION
----

> The VERSION should correspond to the Major.Minor version of the product, whereas MAVEN_VERSION corresponds to the build number.

Reviewer: ```suggestion
+
The `VERSION` corresponds to the major.minor version of {ProductName}, whereas `MAVEN_VERSION` corresponds to the build number.
```

> = Unable to find custom address space plan

Reviewer: ```suggestion
= Unable to find a custom address space plan
```

.Procedure 

> . Check that the address space plan is created in the namespace where {ProductName} is installed. 

Reviewer: ```suggestion
. Ensure that the address space plan was created in the namespace where {ProductName} is installed. 
```

> . Check that the address space plan has passed validation and is in the `Active` phase:

Reviewer: ```suggestion
. Ensure that the address space plan has passed validation and is in the `Active` phase:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} get addressspaceplans -o wide -n {ProductNamespace}
----

> = Unable to find custom address plan

Reviewer: ```suggestion
= Unable to find a custom address plan
```

.Procedure 

> . Check that the address plan is created in the namespace where {ProductName} is installed. 

Reviewer: ```suggestion
. Ensure that the address plan was created in the namespace where {ProductName} is installed. 
```

> . Check that the address plan has passed validation and is in the `Active` phase:

Reviewer: ```suggestion
. Ensure that the address plan has passed validation and is in the `Active` phase:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} get addressplans -o wide -n {ProductNamespace}
----

= Address space is not ready and clients are unable to connect to messaging endpoints

.Procedure

> . Check if address space status message for errors:

Reviewer: ```suggestion
. Check the address space status message for errors:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} get addressspace _myspace_ -o wide
----

. Check address space controller logs for errors:
+
[options="nowrap",subs="+quotes,attributes"]
----

> {cmdcli} logs dc/address-space-controller

Reviewer 2: Should this be `{cmdcli} logs deployment/address-space-controller`?

----

= Addresses are not getting ready

.Procedure

> . Check if address status message for errors:

Reviewer: ```suggestion
. Check the address status message for errors:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} get address _myspace.myaddress_ -o wide
----

> . If using `brokered` address space, check `agent` pod logs for errors:

Reviewer: ```suggestion
. If using the `brokered` address space, check the `agent` pod logs for errors:
```

+
[options="nowrap",subs="+quotes,attributes"]
----

> {cmdcli} logs dc/agent._uuid_

Reviewer 2: Same as above, should this be `deployment` instead deployment config?

----

> . If using `standard` address space, check `admin` pod logs for errors:

Reviewer: ```suggestion
+
If using the `standard` address space, check the `admin` pod logs for errors:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} logs dc/admin._uuid_ -c standard-controller

> {cmdcli} logs dc/admin._uuid_ -c agent

Reviewer 2: Same as above, should these be `deployment` instead deployment config?
